### PR TITLE
fix missing user accounts login window with clear cache

### LINKED ISSUE #4181: With fresh cache, paywall does not properly detect the need for user accounts
To test:

checkout `unlock_provider_base`
clear the localStorage of all unlock-related values.
start the dev environment, with paywall, unlock-app, and locksmith running
go to `http://localhost:3001/static/adremover/integrationtesting-loggedout.html?locks=%5B%7B%22address%22%3A%220x8276A24C03B7ff9307c5bb9c0f31aa60d284375f%22%2C%22%7D%5D&paywall=http%3A%2F%2Flocalhost%3A3001`

You will see the "no account" window instead of the login window

### PR DESCRIPTION
**Pull request by cellog**
# Description

If locks loaded after the initial "show the account iframe" message, the message was ignored. This fixes that race condition.

# Issues

Fixes #4181 

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [x] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [x] My code follows the style guidelines of this project, including naming conventions
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: paywall/src/unlock.js/web3Proxy.ts [hunk 1]
```diff
@@ -121,6 +121,7 @@ export default function web3Proxy(
   let currentLocks: Locks
   let canUseUserAccounts: boolean = !window.web3
   let hasNativeWeb3Wallet = !!window.web3
+  let showIframeWhenReady = false
```
Comment (cnasc) [inline paywall/src/unlock.js/web3Proxy.ts]: Want to make sure I understand correctly:

- `showIframeWhenReady` is only true when we don't have any locks
- when it is true (i.e., we had not previously received locks) and we receive locks, we show the account iframe

Are there any other cases here?

Comment (cellog) [reply paywall/src/unlock.js/web3Proxy.ts]: This is almost correct.

`showIframeReady` is *only* set to true when the `PostMessages.SHOW_ACCOUNTS_MODAL` message is sent before the main window receives locks. In this case, we know the user accounts iframe is ready to receive logins or new account creation requests, but we don't yet know if this paywall has erc20 locks on it or not.

Then, when we get the locks, if there are erc20 locks, it checks this flag, and if set to true, it shows the iframe. Come to think of it, it should then reset it. I'll add that line.

Comment (cellog) [reply paywall/src/unlock.js/web3Proxy.ts]: added

Comment (cnasc) [reply paywall/src/unlock.js/web3Proxy.ts]: Perfect, reshowing the account modal was what I had in mind when asking about other cases

**Pull request merged**